l3.S

```diff
@@ -9,20 +9,26 @@ l3__log_fast:
     syscall
     mov %eax, %fs:0xfffffffffffffffc
 .gotid:
-    mov l3_log(%rip), %rcx      // l3_log -> rcx
+    mov l3_log(%rip), %r8      // l3_log -> r8
 
-    mov  $1, %edx
-    lock xadd %rdx, (%rcx)
-    and $0x3fff, %edx           // idx -> rdx
+    mov  $1, %r9
+    lock xadd %r9, (%r8)
+    and $0x3fff, %r9           // idx -> r9
 
-    add $32, %rcx
-    shl $5, %rdx
-    add %rdx, %rcx
-    mov %rax, (%rcx)  // tid
-    add $8, %rcx
-    mov %rdi, (%rcx)  // msg
-    add $8, %rcx
-    movq $0, (%rcx)  // arg1
-    add $8, %rcx
-    movq $0, (%rcx)  // arg1
+    add $32, %r8
+    shl $5, %r9
+    add %r9, %r8
+    mov %eax, (%r8)  // tid
+#ifdef L3_LOC_ENABLED
```

Comment:
@gregthelaw - There is something funny going on with this logic of stashing away the `__LOC__ ID` value, which should only happen if `L3_LOC_ENABLED=1`. Otherwise, I expect only 0 to be stashed away.

I will explain this as a comment in the top-level UI. One has to xcheck the outputs from CI jobs.

Reply:
I think I figured it out - the L3_LOC_ENABLED macro wasn't finding its way to the .S file. I've pushed a commit to this branch which updates the Makefile.